components/api/src/main/java/com/hotels/styx/api/plugins/spi/PluginMeterRegistry.java

```diff
@@ -0,0 +1,157 @@
+/*
+  Copyright (C) 2013-2020 Expedia Inc.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+ */
+package com.hotels.styx.api.plugins.spi;
+
+import io.micrometer.core.annotation.Incubating;
+import io.micrometer.core.instrument.Counter;
+import io.micrometer.core.instrument.DistributionSummary;
+import io.micrometer.core.instrument.Meter;
+import io.micrometer.core.instrument.MeterRegistry;
+import io.micrometer.core.instrument.Tag;
+import io.micrometer.core.instrument.Tags;
+import io.micrometer.core.instrument.Timer;
+import io.micrometer.core.instrument.search.RequiredSearch;
+import io.micrometer.core.instrument.search.Search;
+import io.micrometer.core.lang.Nullable;
+
+import java.util.Collection;
+import java.util.Map;
+import java.util.function.Consumer;
+import java.util.function.ToDoubleFunction;
+
+public class PluginMeterRegistry {
+    public static final String DEFAULT_TAG_KEY = "plugin";
+    private final MeterRegistry meterRegistry;
+    private final Tag defaultTag;
+    private Tags commonTags;
```

Comment:
Is there a need for this additional set of tags?

Reply:
They were added to give the user the option of adding custom tags that are common to all meters of a plugin.

Reply:
Yes, I understand that, but I couldn't think of an example use case for it. I suppose you could use it to attach the plugin version, maybe. Or perhaps some aspect of the environment that is particular to this instance, that is not captured by Styx's own common tags. Although, Styx doesn't have any common tags, I think - we should probably add some.